docs/core/idioms/parallel.md

```diff
@@ -0,0 +1,95 @@
+# Parallelism in Prefect
```

Comment:
I think we want to expand this idiom into two subsections: "Parallelism within a Flow" and "Parallel Flow execution".  Your write up covers "Parallelism within a Flow" but doesn't address the agent model that allows for parallel flow execution

Reply:
For now let's set this to cover parallelism within a flow. And expand on the sections that have to do with orchestration separately